src/main/webapp/app/exercises/quiz/manage/quiz-exercise-detail.component.ts

```diff
@@ -1,594 +1,101 @@
-            return 'isVisibleBeforeStart';
-        } else if (this.quizExercise.quizEnded) {
-            return 'isOpenForPractice';
-        } else {
-            return 'active';
-        }
-    }
-
-    /**
-     * Returns whether pending changes are present, preventing a deactivation.
-     */
-    canDeactivate(): boolean {
-        return !this.pendingChangesCache;
-    }
-
-    /**
-     * displays the alert for confirming refreshing or closing the page if there are unsaved changes
-     */
-    @HostListener('window:beforeunload', ['$event'])
-    unloadNotification(event: any) {
-        if (!this.canDeactivate()) {
-            event.returnValue = this.translateService.instant('pendingChanges');
-        }
-    }
-
-    /**
-     * @desc Callback for datepicker to decide whether given date should be disabled
-     * All dates which are in the past (< today) are disabled
-     */
-    isDateInPast = (date: NgbDate, current: { month: number }) =>
-        current.month < dayjs().month() + 1 ||
-        dayjs()
-            .year(date.year)
-            .month(date.month - 1)
-            .date(date.day)
-            .isBefore(dayjs());
-
-    /**
-     * Iterates over the questions of the quizExercise and calculates the sum of all question scores
-     */
-    calculateMaxExerciseScore(): number {
-        let scoreSum = 0;
-        this.quizExercise.quizQuestions!.forEach((question) => (scoreSum += question.points!));
-        return scoreSum;
-    }
-
-    /**
-     * Remove question from the quiz
-     * @param questionToDelete {QuizQuestion} the question to remove
-     */
-    deleteQuestion(questionToDelete: QuizQuestion): void {
-        this.quizExercise.quizQuestions = this.quizExercise.quizQuestions?.filter((question) => question !== questionToDelete);
-        this.cacheValidation();
-    }
-
-    /**
-     * Save the quiz to the server and invoke callback functions depending on result
-     */
-    save(): void {
-        if (this.hasSavedQuizStarted || !this.pendingChangesCache || !this.quizIsValid) {
-            return;
-        }
-
-        Exercise.sanitize(this.quizExercise);
-        const filesMap = this.quizQuestionListEditComponent.fileMap;
-        const files = new Map<string, Blob>();
-        filesMap.forEach((value, key) => {
-            files.set(key, value.file);
-        });
-
-        this.isSaving = true;
-        this.quizQuestionListEditComponent.parseAllQuestions();
-        if (this.quizExercise.id !== undefined) {
-            if (this.isImport) {
-                this.quizExerciseService.import(this.quizExercise, files).subscribe({
-                    next: (quizExerciseResponse: HttpResponse<QuizExercise>) => {
-                        if (quizExerciseResponse.body) {
-                            this.onSaveSuccess(quizExerciseResponse.body, true);
-                        } else {
-                            this.onSaveError();
-                        }
+    getExerciseDetailSections() {
```

Comment:
```suggestion
    private getExerciseDetailSections() {
```

This can be private.